### ISSUE TITLE
Wait block needed in loops

### ISSUE THREAD

**Issue by gepponline** [thunkable/thunkable-issues#915]
In ThunkableX I'm using a count block to clone a row and change label text for each clone incrementing a number at the end of the label.

In live testing and preview, the name of the labels is still the same of the parent If I don't insert a wait for 0 sec inside the loop.

To reproduce it:
1. Create a Button and a column, inside the column place a row and inside the row place a label with text Manutenzione_1.
2. Run this code:
![image](https://user-images.githubusercontent.com/52273816/118927899-c04a3180-b942-11eb-87f0-90c27713e26c.png)

The result will be 5 labels in column, all with "Manutenzione_1" as text
3. If you run this code, with a 0 seconds wait block, instead:
![image](https://user-images.githubusercontent.com/52273816/118928069-fe475580-b942-11eb-9d86-673eecd69dd0.png)

The result will be 5 labels in column with "Manutenzione_1", "Manutenzione_2","Manutenzione_3" and so on as text, as expected.

**Comment by tatiang** [comment]
I've made it standard practice to add a "wait 0 seconds" block in any loops that use clone blocks. I still don't know why that's needed but it does seem to make a difference.

**Comment by Addie924** [comment]
Thanks for posting this! I'll bear this in mind the next time I clone inside a loop. Does this hold true for 'for each item j' loops as well?

**Comment by gepponline** [comment]
Yes, same beahavior with for loops as well.

**Comment by jane-d-alt** [comment]
Hi there, thanks for reporting this behavior. Can you include a project link that executes both of these blocks of code? Having an app to demonstrate the behavior reported will make it easier to file an internal report and get this behavior fixed. Thanks!

**Comment by gepponline** [comment]
Here it is.
https://x.thunkable.com/projects/60ab433e0f9e9500115e6092/ecfc09fa-9812-406b-a6dd-9f75377cd047/blocks
As you can see, without wait block, all label still remain the same as the original row.
With the wait block, each label assume a new value as expected

**Comment by tatiang** [comment]
Yep, your project makes it really clear. The wait block is needed to give the clone block enough time to complete before it's called again in the loop. The wait 0 seconds block can go above or below the label set block but it has to be within the clone block's "do" section.
